ister_gui.py

```diff
@@ -2395,10 +2474,18 @@ def _save_config(self, config, mount_d):
                     output = ''
                 if 'TYPE="' in output:
                     config['FilesystemTypes'][-1]['disable_format'] = True
+            if "encryption" in mount_d[point]:
+                config['FilesystemTypes'][-1]["encryption"] = {
+                    "passphrase": mount_d[point]['encryption'],
+                    "name" : "encrypted_dev" }
             config['PartitionMountPoints'].append({
                 'disk': disk,
                 'partition': part,
                 'mount': point})
+            if "encryption" in mount_d[point]:
+                config['PartitionMountPoints'][-1]["encryption"] = {
```

Comment:
So thinking about this a bit, I'd like to know how you would want to handle multiple partitions and disks being used. Do you expect all of them on a system to share the same encryption key or should they be able to have different keys per partition? I'm actually not sure if there is a use case that makes much sense to me to support from the installer for different keys so I was thinking about having Encryption be an optional toplevel item in the template with a passphrase and a list of dictionaries with disk + partition in them. Thoughts?

Reply:
I was thinking in the screen when you decide if a partition will be formatted, just add there the option for encrypt and if the type of partition is /root apply this process if not add information to the crypttab but this could be a future improvement, at this time I am just handling root partitioning, crypttab support hash for passphrase then every partition could have a passphrase